pyaedt/filtersolutions_core/ideal_response.py

```diff
@@ -0,0 +1,458 @@
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from ctypes import POINTER
+from ctypes import byref
+from ctypes import c_bool
+from ctypes import c_char_p
+from ctypes import c_double
+from ctypes import c_int
+from enum import Enum
+import math
+
+import pyaedt
+from pyaedt.filtersolutions_core.graph_setup import GraphSetup
+
+
+class FrequencyResponseColumn(Enum):
+    """Provides an enum of frequency response parameters.
+
+    Attributes:
+    MAGNITUDE_DB: Represents the frequency response magnitude in dB.
+    PHASE_DEG: Represents the frequency response phase in degree.
+    GROUP_DELAY: Represents the frequency response group delay.
+    PHASE_RAD: Represents the frequency response phase in radian.
+    MAGNITUDE_ARITH: Represents the frequency response magnitude.
+    MAGNITUDE_REAL: Represents the real part of frequency response  magnitude .
+    MAGNITUDE_IMAG: Represents the imaginary part of frequency response  magnitude.
+    PHASE_DEV_DEG: Represents the frequency response phase deviation in degree.
+    PHASE_DEV_RAD: Represents the frequency response phase deviation in radian.
+    FREQUENCY: Represents the frequency response frequency parameter.
+    """
+
+    MAGNITUDE_DB = 0
+    PHASE_DEG = 1
+    GROUP_DELAY = 2
+    PHASE_RAD = 4
+    MAGNITUDE_ARITH = 3
+    MAGNITUDE_REAL = 5
+    MAGNITUDE_IMAG = 6
+    PHASE_DEV_DEG = 7
+    PHASE_DEV_RAD = 8
+    FREQUENCY = 9
+
+
+class TimeResponseColumn(Enum):
+    """Provides an enum of time response parameters.
+
+    Attributes:
+    STEP_RESPONSE: Represents the step time response.
+    RAMP_RESPONSE: Represents the ramp time response.
+    IMPULSE_RESPONSE: Represents the impulse time response.
+    STEP_RESPONSE_DB: Represents the step time response in dB.
+    RAMP_RESPONSE_DB: Represents the ramp time response in dB.
+    IMPULSE_RESPONSE_DB: Represents the impulse time response in dB.
+    TIME: Represents time response the time parameter.
```

Comment:
```suggestion
    **Attributes:**
    
    - STEP_RESPONSE: Represents the step time response.
    - RAMP_RESPONSE: Represents the ramp time response.
    - IMPULSE_RESPONSE: Represents the impulse time response.
    - STEP_RESPONSE_DB: Represents the step time response in dB.
    - RAMP_RESPONSE_DB: Represents the ramp time response in dB.
    - IMPULSE_RESPONSE_DB: Represents the impulse time response in dB.
    - TIME: Represents time response the time parameter.
```
The last item seems to be missing a word.